scipy/special/tests/test_mpmath.py

```diff
@@ -49,6 +49,22 @@ def test_expi_complex():
 
     FuncData(sc.expi, dataset, 0, 1).check()
 
+#------------------------------------------------------------------------------
+# hyp0f1
+#------------------------------------------------------------------------------
+
+def test_hyp0f1_gh5764():
+    # Do a small and somewhat systematic test that runs quickly
+    pts = []
+    axis = [-99.5, -9.5, -0.5, 0.5, 9.5, 99.5]
+    for v in axis:
+        for x in axis:
+            for y in axis:
+                pts.append((v, x + 1J*y))
+    std = np.array([(complex(mpmath.hyp0f1(*p)),) for p in pts])
+    # Can't use FuncData because v should be real and z complex
```

Comment:
This was usually solved by defining the points as complex, and then defining a new function that takes complex arguments but discards the imaginary part.